daemon/config/config_test.go

```diff
@@ -153,6 +153,32 @@ func TestDaemonConfigurationMergeConflictsWithInnerStructs(t *testing.T) {
 	}
 }
 
+// Test for #40711
+func TestDaemonConfigurationMergeFlagsWitchConfigFile(t *testing.T) {
```

Comment:
Typo 😅 

```suggestion
func TestDaemonConfigurationMergeFlagsWithConfigFile(t *testing.T) {
```

Perhaps rename to include the DefaultAddressPools ?

```suggestion
func TestDaemonConfigurationMergeDefaultAddressPools(t *testing.T) {
```